lib/smart_proxy_remote_execution_ssh/api.rb

```diff
@@ -4,6 +4,120 @@ class Api < ::Sinatra::Base
       get "/pubkey" do
         File.read(Ssh.public_key_file)
       end
+      post "/session" do
+        if env["HTTP_CONNECTION"] != "upgrade" or env["HTTP_UPGRADE"] != "raw"
+          return [ 401, "Invalid request: /ssh/session requires connection upgrade to 'raw'" ]
+        end
+        socket = nil
+        if env['rack.hijack?']
+          env['rack.hijack'].call
+          socket = env['rack.hijack_io']
+        end
+        if !socket
+          return [ 501, "Internal error: request hijacking not available" ]
```

Comment:
`unless $CONDITION` is preferred over `if !$CONDITION`, This whole block could be rewritten as

```ruby
return [501, "Internal error: request hijacking not available"] unless socket
```

On a side note, spaces just after the opening bracket (and just before the closing one) are not really all that common in ruby world